src/formatDate.js

```diff
@@ -8,7 +8,60 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[3]);
+  const obj = {};
+  const tab = [];
+  let year;
+
+  const ifFromIncludesYY = fromFormat.includes('YY');
+  const ifFromIncludesYYYY = fromFormat.includes('YYYY');
+  const ifToIncludesYY = toFormat.includes('YY');
+  const ifToIncludesYYYY = toFormat.includes('YYYY');
+
+  const fromIndexYY = fromFormat.indexOf('YY');
+  const fromIndexYYYY = fromFormat.indexOf('YYYY');
+
+  if (ifFromIncludesYY && ifToIncludesYYYY) {
+    fromFormat[fromIndexYY] = 'YYYY';
```

Comment:
Its really bad practice to mutate input values. Ensure that any nested objects are also copied (deep copy).